### PR TITLE
Separate out python dependencies for running scripts

### LINKED ISSUE #28580: [CERT-TEST-FAILURE] Python cert scripts in `src/python_testing` doesn't document dependencies.
### Feature Area

Other

### Test Case

TC-CGEN-2.4, TC-TIMESYNC-2.2, TC-TIMESYNC-2.4, TC-TIMESYNC-2.5, TC-TIMESYNC-2.6, TC-TIMESYNC-2.7, TC-TIMESYNC-2.8, TC-TIMESYNC-2.9, TC-SC-3.6, TC-DA-1.7, TC-ACE-1.3, TC-ACE-1.4, TC-RR-1.1

### Reproduction steps

On Matter Test Harness setup run:

Run  docker container with `chip-cert-bins` image:
```sh
docker run -v /var/paa-root-certs:/paa_roots -v /var/run/dbus/system_bus_socket:/var/run/dbus/system_bus_socket -v /launch_dir --privileged --network host -it connectedhomeip/chip-cert-bins:5b4f8004662d00bdb111367fec7d3ea978c23372 bash
```

Run test script inside docker:
```sh
rm -f admin_storage.json && python3 python_testing/TC_RR_1_1.py --discriminator 20202021 --passcode 3842 --commissioning-method on-network --paa-trust-store-path /paa_roots --storage-path admin_
storage.json
```

Notice that test script failed due to missing dependency. `mobly`

### Bug prevalence

Always

### GitHub hash of the SDK that was being used

5b4f8004662d00bdb111367fec7d3ea978c23372

### Platform

raspi

### Anything else?

Suggested fix by @andreilitvin  and @mikaelhm:

- Add `requirements.txt` to [src/python_testing](https://github.com/project-chip/connectedhomeip/tree/master/src/python_testing) folder. And migrate away from using `scripts/setup/requirements.python_tests.txt`.
- Add `README.md` to [src/python_testing](https://github.com/project-chip/connectedhomeip/tree/master/src/python_testing) documenting how to use those tests and how to install requirements manually
- `chip-cert-bins` Dockerfile should install python dependencies using the new `requirements.txt` file.

### PR DESCRIPTION
**Pull request by andy31415**
Changes:
  - add dependencies to py_matter_idl instead of relying on requirements.txt
  - add separate requirements.txt for scripts/tests and src/python_testing
  - update Dockerfile for chip-cert-bins

I expect this fixes #28580

### CODE REVIEW

**Review by mikaelhm — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Comment by andy31415** [comment]
Fast track: CSG + 1 more approval. This should only affect building and docker images, not functionality.

File: third_party/imgui/repo
Comment (bzbarsky-apple) [inline third_party/imgui/repo:1]: All these submodule changes look unrelated.....

Comment (andy31415) [reply third_party/imgui/repo:1]: Not intended. Thank you for catching.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by mikaelhm — APPROVED** [review]
